bluebrain/repo-patches/packages/py-pint-xarray/package.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyPintXarray(PythonPackage):
+    """Physical units interface to xarray using Pint."""
+
+    homepage = "https://github.com/xarray-contrib/pint-xarray"
+    pypi = "pint-xarray/pint-xarray-0.2.1.tar.gz"
+
+    version('0.2.1', sha256='f4d0caa713239e6847a7c6eefe2427358566451fe56497d533f21fb590a3f313')
```

Comment:
I think I found the problem.
This hash is wrong and it's causing to install pint 0.17 instead of pint-xarray 0.2.1 (it's the hash of pint 0.17).
With the correct hash `1ee6bf74ee7b52b946f226a96469276fa4f5c68f7381c1b2aae66852562cb275`, the right package is installed.
I'm very surprised that the spack installation doesn't fail when using the hash of a different package.

Reply:
Yep, I was also very surprised when discovering that long time ago but totally forgot about this now... :facepalm: 
Many thanks @GianlucaFicarelli !!!
rebuilding with new hash...

Reply:
> I'm very surprised that the spack installation doesn't fail when using the hash of a different package.

`morphio` and `py-morphio` share the same tarball + hash, resources may similarly. So in some sense resolving stuff by the hash specified by the user makes sense. OTOH, I'd kind of expect that Spack would hash the URL/pypi in the source cache too. This whole thing has also bitten me at times.

Reply:
Maybe it's worth to open an issue with the upstream? I would expect that a checksum is used to verify that spack is downloading the exact same package, not that it's used to decide which package to download, regardless of the package name (and maybe the version).

Reply:
@matz-e you are right that the spack packages `morphio` and `py-morphio` use the same tarball + hash, but they use the same `url` and `git` parameters as well. Only the final spack package name is different.
In every case that I can think, url+version+hash should match together.

Reply:
> not that it's used to decide which package to download

This is coming from the cache, not download. But I agree that this could be handled better.